coa-helm/templates/cbas-api.yaml

```diff
@@ -1,24 +1,29 @@
-{{ include "terra-batch-libchart.cbas-api-deploy" (list . "coa-helm.cbas-api-deploy") }}
 {{ define "coa-helm.cbas-api-deploy" }}
+{{- if .Values.cbas.coaEnabled }}
```

Comment:
I think this pattern might work instead:
```
{{- if .Values.cbas.coaEnabled }}
{{ include "terra-batch-libchart.cbas-api-deploy" (list . "coa-helm.cbas-api-deploy") }}
{{ end }}
{{ define "coa-helm.cbas-api-deploy" }}
# Overrides go here
spec:
  template:
    metadata:
      labels:
    {{ include "coa-helm.labels" $ | nindent 8 }}
{{ end }}
```

This way the definition (function) is still there -- but is gated from executing via the `if` block.

Reply:
Unfortunately, I don't think the syntax of Helm templates allows us to have an `if` block at the top, rather, it must be a `define`

See here for more details: https://broadinstitute.slack.com/archives/C02GJ782BAA/p1674758933800709?thread_ts=1674756048.634379&cid=C02GJ782BAA

Reply:
(Here is the direct Github explanation as well: https://github.com/helm/helm/issues/10395#issuecomment-978104760)

Reply:
The above works for me in `helm template`.

I think that issue is different -- here the `define` isn't _contained_ in the if, they are independent. (You could do define..end then if..end if you want).

The define doesn't do anything by itself -- the `include` is what actually adds the template (and merges it with the base chart).

Reply:
Updated!